src/components/FileUploaderV2/FileUploaderV2.js

```diff
@@ -0,0 +1,257 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import Icon from '../Icon';
+import uid from '../../tools/uniqueId';
+import { ButtonTypes } from '../../prop-types/types';
+import { iconCloseSolid, iconCheckmarkSolid } from 'carbon-icons';
+
+export class FileUploaderButtonV2 extends Component {
+  static propTypes = {
+    className: PropTypes.string,
+    labelText: PropTypes.string,
+    multiple: PropTypes.bool,
+    name: PropTypes.string,
+    onChange: PropTypes.func,
+    onClick: PropTypes.func,
+    role: PropTypes.string,
+    tabIndex: PropTypes.number,
+    buttonKind: ButtonTypes.buttonKind,
+    accept: PropTypes.arrayOf(PropTypes.string),
+  };
+  static defaultProps = {
+    tabIndex: 0,
+    labelText: 'Add file',
+    buttonKind: 'primary',
+    multiple: false,
+    onChange: () => {},
+    onClick: () => {},
+    accept: [],
+  };
+  uid = this.props.id || uid();
+
+  render() {
+    const {
+      className,
+      multiple,
+      role,
+      tabIndex,
+      buttonKind,
+      accept,
+      name,
+      onChange,
+      labelText,
+      ...other
+    } = this.props;
+    const classes = classNames({
+      'bx--file': true,
+      [className]: className,
+    });
+
+    return (
+      <div
+        role="button"
+        tabIndex="0"
+        className={classes}
+        onKeyDown={evt => {
+          if (evt.which === 13 || evt.which === 32) {
+            this.input.click();
+          }
+        }}>
+        <label
+          className={`bx--btn bx--btn--${buttonKind}`}
+          tabIndex={tabIndex}
+          htmlFor={this.uid}
+          role={role}
+          {...other}>
+          {labelText}
+        </label>
+        <input
+          className="bx--visually-hidden"
+          ref={input => (this.input = input)}
+          id={this.uid}
+          type="file"
+          multiple={multiple}
+          accept={accept}
+          name={name}
+          onChange={onChange}
+          onClick={evt => {
+            evt.target.value = null;
+          }}
+        />
+      </div>
+    );
+  }
+}
+
+export function FilenameV2({
+  iconDescription,
+  onKeyDown = () => {},
+  status = 'uploading',
+  style = {},
+  tabIndex = 0,
+  ...other
+}) {
+  switch (status) {
+    case 'uploading':
+      return (
+        <div
+          className="bx--loading"
+          style={{ ...style, width: '1rem', height: '1rem' }}
```

Comment:
would it be better to modify [these CSS rules](https://github.com/IBM/carbon-components/blob/master/src/components/file-uploader/_file-uploader.scss#L75-L76) in `_file-uploader.scss` rather than defining a new CSS class?

Reply:
Great question @emyarod! Sounds that it makes sense unless change in style there breaks vanilla.

Reply:
hi @asudoh, I've made a PR in the vanilla repo so that we can remove these inline styles https://github.com/IBM/carbon-components/pull/1016

Reply:
Hi @emyarod thank you for doing this! Seems that the PR contains a breaking change - Tapping designers to see if such breaking change can be done now. Thinking that introducing React-specific CSS class may be an option.

Reply:
the inline styles are removed now that the vanilla repo has been modified @asudoh